scripts/pre-commit.sh

```
@@ -0,0 +1,343 @@
+  echo -e "\n${BOLD}${RED}Failed checks and their errors:${NC}"
+  
+  for i in "${!FAILED_STEPS[@]}"; do
+    echo -e "\n${RED}======================================${NC}"
+    echo -e "${RED}    Failed check: ${FAILED_STEPS[$i]}    ${NC}"
+    echo -e "${RED}======================================${NC}"
+    
+    # Display the first 15 lines of the error log
+    head -n 15 "${FAILED_LOGS[$i]}"
+    
+    # Check if there's more content and show a message
+    if [ $(wc -l < "${FAILED_LOGS[$i]}") -gt 15 ]; then
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)